service/controller/resource/namespace/delete.go

```diff
@@ -13,7 +14,7 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	_, err = r.k8sClient.K8sClient().CoreV1().Namespaces().Create(namespace)
+	err = r.k8sClient.K8sClient().CoreV1().Namespaces().Delete(namespace.GetName(), &metav1.DeleteOptions{})
 	if apierrors.IsAlreadyExists(err) {
```

Comment:
This isn't right either, right?

Reply:
Do you mean the error should be `NotFound`? That's right, fixing this.